src/backend/RecentImagesStore.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2021 Damir Porobic <damir.porobic@gmx.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "RecentImagesStore.h"
+
+
+RecentImagesStore::RecentImagesStore()
+{
+	loadRecentImages();
+}
+
+void RecentImagesStore::loadRecentImages()
+{
+	const auto numRecentImages = mSettings.beginReadArray("recentImages");
+	for (int i=0; i<numRecentImages; ++i) {
+		mSettings.setArrayIndex(i);
+		const auto recentImagePath = mSettings.value("imagePath").toString();
+		mRecentImages.enqueue(recentImagePath);
+	}
+	mSettings.endArray();
+}
+
+void RecentImagesStore::storeImage(const QString &image)
+{
+	static const uint maxRecentImages = 10;
+
+	if (mRecentImages.contains(image)) {
```

Comment:
As soon as you save the image, it will add it to the list of recently opened images. If you then modify the image and save it again, it will not add it to the list again; it will only add it to the list if you save it under a different name / location (it's a new file now).

Reply:
Now I see, the name confused me, I was not looking on the parameter that it's actually a string, just saw the Name image. Where I come from they say "Pretpostavka je majka zajeba" which translates to something like "Assumption is the mother of fuck ups".

Reply:
That's a good saying, will keep it in mind. Well, the English translation at least :D